examples/msr-events.rs

```diff
@@ -0,0 +1,111 @@
+use clap::{App, Arg, ArgMatches};
+use colored::*;
+use env_logger;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+use std::time::Instant;
+use std::u32;
+
+use microvmi::api::{EventReplyType, EventType, InterceptType, Introspectable};
+
+fn parse_args() -> ArgMatches<'static> {
+    App::new(file!())
+        .version("0.3")
+        .author("Mathieu Tarral")
+        .about("Watches msr register VMI events")
+        .arg(Arg::with_name("vm_name").index(1).required(true))
+        .arg(
+            Arg::with_name("register")
+                .multiple(true)
+                .takes_value(true)
+                .short("r")
+                .default_value("0x174"),
+        )
+        .get_matches()
+}
+
+fn toggle_msr_intercepts(drv: &mut Box<dyn Introspectable>, vec_msr: &Vec<u32>, enabled: bool) {
+    drv.pause().expect("Failed to pause VM");
+
+    for msr in vec_msr {
+        let intercept = InterceptType::Msr(*msr);
+        let status_str = if enabled { "Enabling" } else { "Disabling" };
+        println!("{} intercept on 0x{:x}", status_str, msr);
+        for vcpu in 0..drv.get_vcpu_count().unwrap() {
+            drv.toggle_intercept(vcpu, intercept, enabled)
+                .expect(&format!("Failed to enable 0x{:x}", msr));
+        }
+    }
+
+    drv.resume().expect("Failed to resume VM");
+}
+
+fn main() {
+    env_logger::init();
+
+    let matches = parse_args();
+
+    let domain_name = matches.value_of("vm_name").unwrap();
+    let registers: Vec<_> = matches.values_of("register").unwrap().collect();
```

Comment:
I think it's not really a good example to just pass integers as msr indices for two reasons:
1. You have to look those indices up.
2. Currently we are only supporting a very limited set of those indices. The user would have to trial an error to find out which indices actually work.
Therefore I suggest that you match strings like "lstar" to their respective indices. We only support 6 msr values currently so it's not hard to do. Additionally you should write what values are possible into the argument description.